core/camera/src/main/java/com/google/jetpackcamera/core/camera/effects/processors/LowLightBoostSurfaceProcessor.kt

```diff
@@ -0,0 +1,198 @@
+    private var lowLightBoostSession: LowLightBoostSession? = null
+    private var inputSurfaceRequestScope: SurfaceRequestScope? = null
+
+    // Executor for provideSurface callback
+    private val glExecutor = Executors.newSingleThreadExecutor()
+
+    init {
+        coroutineScope.launch(start = CoroutineStart.UNDISPATCHED) {
+            outputSurfaceFlow
+                .filterNotNull()
+                .collectLatest { surfaceOutputScope ->
+                    surfaceOutputScope.withSurfaceOutput { refCountedSurface,
+                                                           size,
+                                                           updateTransformMatrix ->
+                        // If we can't acquire the surface, then the surface output is already
+                        // closed, so we'll return and wait for the next output surface.
+                        val outputSurface =
+                            refCountedSurface.acquire() ?: return@withSurfaceOutput
+
+                        tryCreateLowLightBoostSession(outputSurface)
+                    }
+                }
+        }
+    }
+
+    private fun createLowLightBoostCallback(): LowLightBoostCallback =
+        object : LowLightBoostCallback {
+            override fun onSessionDestroyed() {
+                Log.d(TAG, "LLB session destroyed")
+                releaseLowLightBoostSession()
+            }
+
+            override fun onSessionDisconnected(status: Status) {
+                Log.d(TAG, "LLB session disconnected: $status")
+                releaseLowLightBoostSession()
+            }
+        }
+
+    @RequiresApi(Build.VERSION_CODES.R)
+    override fun onInputSurface(request: SurfaceRequest) {
+        // Cancel any previous request handling
+        inputSurfaceRequestScope?.cancel("New SurfaceRequest received.")
+        val newScope = SurfaceRequestScope(request) // Helper to manage request lifecycle
+        inputSurfaceRequestScope = newScope
+
+        outputSurfaceFlow.value?.surfaceOutput?.let { outputSurface ->
+            coroutineScope.launch {
+                val surface = outputSurface.getSurface(glExecutor, {})
+                tryCreateLowLightBoostSession(surface)
+            }
+        }
+
+        // Clean up when the request is released (e.g., camera is closed)
+        request.addRequestCancellationListener(glExecutor) { // Replace context if not available
+            Log.d(TAG, "InputSurfaceRequest cancelled: $request")
+            inputSurfaceRequestScope?.cancel("SurfaceRequest cancelled by CameraX.")
+            inputSurfaceRequestScope = null
+            releaseLowLightBoostSession()
+        }
+    }
+
+    @RequiresApi(Build.VERSION_CODES.R)
+    private suspend fun tryCreateLowLightBoostSession(outputSurfaceForLlb: Surface) {
+        inputSurfaceRequestScope?.withSurfaceRequest { currentInputRequest ->
+
+            coroutineScope.launch { // Launch in your processor's scope
```

Comment:
I don't think this launch is necessary. This code is already within a private suspend fun that is called within a flow collector. It will get cancelled if a new `SurfaceOutput` comes in, and is already within the correct scope.

Reply:
Done